### DISCUSSION TITLE
Create XML metadata file to accompany reveal.js presentation made in Quarto

### DISCUSSION THREAD

**Discussion by erichgoldman** [quarto-dev/quarto-cli#2323] [Q&A]
I am looking to automate some steps I perform manually directly through Quarto + pandoc.

I am creating `reveal.js` presentations. For each presentation, I would like to generate a customized XML metadata file which will include document metadata and then a list of all files, scripts, etc and live in the folder root. I would prefer the ability to create an XML template or use XSL versus combining multiple string concatenations. 

Ideally, the end result would also be a zip folder instead of creating that manually (i.e., quarto will dump out the final product I need), but if that is too much that is okay. I want to eventually hand this off to less technical people and want to avoid manual steps or other dependencies as much as possible.

It also wasn't clear to me if multiple formats would go to their own folders or the same output folder.

I've been poking around the docs, but I am not sure where to get started exactly. I see some snippets that make sense, but I am missing how to patch everything together.

---

I think this functionality would be similar to [[OPML](https://quarto.org/docs/reference/formats/opml.html)](https://quarto.org/docs/reference/formats/opml.html), but I couldn't figure out where the source code is for this on GitHub.

From the Quarto docs, it seems I can create a Lua filter to accomplish this as well. I found the example to "[[create a handout from a paper](https://pandoc.org/lua-filters.html#creating-a-handout-from-a-paper)](https://pandoc.org/lua-filters.html#creating-a-handout-from-a-paper)", but I couldn't see how you get from that `pandoc` object to a specific document type/output from the documentation, where you call from, how you write, etc. An end-to-end example for something like this through Quarto would be helpful.

Some of the key things I want to do.

- Create XML file that lives in the root directory of the reveal.js html output
- In the XML document:
  - Include metadata about the presentation like title, author, etc.
  - Include a list of all output files, scripts, images, etc. for continuity checks
  - Reference each slide, such as listing the title for each slide and other metadata I may define at the slide level (property/class/etc). [This is in the document AST?]

I want to minimize dependencies, so I would like to make this a Lua based Quarto filter. If there are recommended libraries for doing some of this stuff with Lua please provide suggestions for those as well. Any references or example code / repo would be very helpful.

**Comment by cderv** [comment]
Hi, 

Just wanted to share hints about your search. 

> I think this functionality would be similar to [[OPML](https://quarto.org/docs/reference/formats/opml.html)]([quarto.org/docs/reference/formats/opml.html](https://quarto.org/docs/reference/formats/opml.html)), but I couldn't figure out where the source code is for this on GitHub.

OPML is available with Quarto because it is a supported output format in Pandoc. As far as I know Quarto does nothing special for this output format in addition to Pandoc support. 

Regarding Lua filter, our doc is here: https://quarto.org/docs/extensions/lua.html It gives some resources and helps on how to write filters for Pandoc. 

Also, it is quite advanced Pandoc usage, but you can write custom writers too with Pandoc : https://pandoc.org/custom-writers.html 
which should be possible to use with Quarto too.

So no solution, but hopefully some useful resourcein.
